### PR TITLE
[ODS-4644] Create Profiles NuGet package

### PR DESCRIPTION
**Pull request by semalaiappan**

### CODE REVIEW

File: Application/EdFi.Ods.Api/Conventions/TemplateSetConventions.cs [hunk 1]
```diff
@@ -0,0 +1,13 @@
+﻿// SPDX-License-Identifier: Apache-2.0
+// Licensed to the Ed-Fi Alliance under one or more agreements.
+// The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+// See the LICENSE and NOTICES files in the project root for more information.
+
+namespace EdFi.Ods.Api.Conventions
+{
+    public static class TemplateSetConventions
```
Comment (gmcelhanon) [inline Application/EdFi.Ods.Api/Conventions/TemplateSetConventions.cs:8]: This same file was on the PR for ODS-4664, so I'm copying the comment here as well.

"TemplateSet" is terminology copied in from the CodeGen utility. I think `PluginConventions` would be a better name for this class in this context.

File: Application/EdFi.Ods.Api/Helpers/AssemblyLoaderHelper.cs [hunk 2]
```diff
@@ -142,35 +147,34 @@ public static IEnumerable<string> FindPluginAssemblies(string pluginFolder, bool
...
+                if (Directory.Exists(extensionFolder))
                 {
-                    string extensionFolder = Path.GetDirectoryName(assemblyPath);
-                    var validationResult = validator.ValidateObject(extensionFolder);
-
-                    if (!validationResult.Any())
+                    string[] assemblyMetadataPath = Directory.GetFiles(
+                        extensionFolder, AssemblyMetadataSearchString, SearchOption.AllDirectories);
+                        
+                    if (assemblyMetadataPath.Length>0 && IsExtensionInAssemblyModelType(assemblyMetadataPath[0]))
```
Comment (gmcelhanon) [inline Application/EdFi.Ods.Api/Helpers/AssemblyLoaderHelper.cs:157]: This looks to be based on an earlier version of https://github.com/Ed-Fi-Alliance-OSS/Ed-Fi-ODS/pull/247 for ODS-4664 and needs to be rebased on that after the appropriate changes are made there.

**Review by gmcelhanon — CHANGES_REQUESTED** [review]
This looks to be based on an earlier version of the branch for https://github.com/Ed-Fi-Alliance-OSS/Ed-Fi-ODS/pull/247 (for ODS-4664) and needs to be rebased on that after the appropriate changes are made there.

**Comment by semalaiappan** [comment]
@gmcelhanon  Ready for review again

**Pull request closed without merging**